vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -46,4 +54,27 @@ public ISlide getSlide(String slideId) {
     public void updateSlide(Slide slide) {
         slideRepo.save((Slide) slide);
     }
+    
+    @Override
+    public void deleteSlideById(String slideId) throws SlideDoesNotExistException {
+        try {
+            slideRepo.delete((Slide) getSlide(slideId));
+        } catch (IllegalArgumentException | EmptyResultDataAccessException exception) {
+            throw new SlideDoesNotExistException(exception);
+        }
+    }
+
+	@Override
+	public void deleteSlidesFromSequence(String slideId, String moduleId) throws SlideDoesNotExistException {
+
+		List<Sequence> sequences = sequenceRepo.findSequencesForModule(moduleId);
+		// Remove slides that refer Sequence
+		for (int i = 0; i < sequences.size(); i++) {
+			if (sequences.get(i).getSlides().get(i).getId().equals(slideId)) {
```

Comment:
what is this doing? why are you only looking at the i'th slide of any i'th sequence?

Reply:
The sequences is the total sequences belonging to a module. So to delete reference of a slide belonging to that specific sequence, I have to loop through it.

Reply:
I'm confused. "So to delete reference of a slide belonging to that specific sequence", what specific sequence? Right now you take the first sequence and check if the first slide has a specific slide id, then you take the second sequence and check if the second slide has a specific slide id, then you take the third sequence and its third slide, etc. Why?

Reply:
Because, findSequencesForModule() is giving me all the sequences belonging to the module and not Slide. 
Hence, I am looping each sequence object from a list of sequences object and matching it with the selected slide, to delete it.